tutorials/settings_tutorial.md

```diff
@@ -0,0 +1,129 @@
+As discussed in [Implementing the PersistentStateComponent Interface](/basics/persisting_state_of_components.md#implementing-the-persistentstatecomponent-interface), `AppSettingsState` uses the pattern of implementing [`PersistentStateComponent`](upsource:///platform/projectModel-api/src/com/intellij/openapi/components/PersistentStateComponent.java) itself:
+
+```java
+{% include /code_samples/settings/src/main/java/org/intellij/sdk/settings/AppSettingsState.java %}
+```
+
+#### Storage Macro
+The [`@State`](upsource:///platform/projectModel-api/src/com/intellij/openapi/components/State.java) macro, located just above the class declaration, [defines the data storage location](/basics/persisting_state_of_components.md#defining-the-storage-location).
+For `AppSettingsState`, the data `name` parameter is the FQN of the class.
+Using FQN is a best practice to follow, and is required if custom data gets stored in the standard project or workspace files.
+
+The `storages` parameter utilizes the [`@Storage`](upsource:///platform/projectModel-api/src/com/intellij/openapi/components/Storage.java) macro to define a custom file name for the `AppSettingsState` data.
+In this case, the file is located in the `options` directory of the [configuration directory](https://www.jetbrains.com/help/idea/tuning-the-ide.html?_ga=2.150184054.1363126807.1588191212-22652347.1575391521#config-directory) for the IDE.
+
+#### Persistent Data Fields
+The `AppSettingState` implementation has two public fields: a `String` and a `boolean`.
+Conceptually these fields hold the name of a user, and whether that person is an IntelliJ IDEA user, respectively.
+See [Implementing the State Class](/basics/persisting_state_of_components.md#implementing-the-state-class) for more information about how `PersistentStateComponent` serializes public fields.
+
+#### AppSettingState Methods
+The fields are so limited and straightforward for this class that encapsulation is not used for simplicity.
+All that's needed for functionality is to override the two methods called by the IntelliJ Platform when a new component state is loaded (`PersistentStateComponent.loadState()`), and when a state is saved (`PersistentStateComponent.getState()`). 
+See `PersistentStateComponent` for more information about these methods. 
+
+One static convenience method has been added - `AppSettingState.getInstance()` - which allows `AppSettingsConfigurable` to easily acquire a reference to `AppSettingState`.
+
+## The AppSettingsComponent Class
+The role of the [`AppSettingsComponent`](https://github.com/JetBrains/intellij-sdk-docs/tree/master/code_samples/settings/src/main/java/org/intellij/sdk/settings/AppSettingsComponent.java) is to provide a `JPanel` for the custom Settings to the IDE Settings Dialog.
+The `AppSettingsComponent` has-a `JPanel`, not shown in the class diagram above, and is responsible for its lifetime.
+No Extension Point declaration is required for `AppSettingsComponent`.
```

Comment:
"No Extension Point declaration is required .." remove, replace with "is instantiated from AppSettingsConfigurable#createComponent"